components/list/list-item-checkbox-mixin.js

```diff
@@ -0,0 +1,85 @@
+import { css, html } from 'lit-element/lit-element.js';
+import { checkboxStyles } from '../inputs/input-checkbox-styles.js';
+import { getUniqueId } from '../../helpers/uniqueId.js';
+import { nothing } from 'lit-html';
+
+export const ListItemCheckboxMixin = superclass => class extends superclass {
+
+	static get properties() {
+		return {
+			disabled: {type: Boolean },
+			key: { type: String, reflect: true },
+			selectable: {type: Boolean },
+			selected: { type: Boolean, reflect: true }
+		};
+	}
+
+	// TODO: Remove grid styles after list-item-generic is created
+	static get styles() {
+		return [ checkboxStyles, css`
+		[slot="control"] {
```

Comment:
If I understand correctly, use of the checkbox mixin requires that the component defines the named gridlines.  I think I'd prefer it if the checkbox mixin didn't need to know anything about the layout, and perhaps this could be done by the mixin defining two render helpers: one that renders the checkbox, and one that renders the toggle area, and let the consuming component compose the layout however it needs to.
```
renderCheckbox() {
    return html`
        <input
            id="${this._checkboxId}"
            class="d2l-input-checkbox"
            @change="${this._handleCheckboxChange}"
            type="checkbox"
            .checked="${this.selected}"
            ?disabled="${this.disabled}">
    `;
}

renderCheckboxToggleContainer(inner) {
    return html`<div @click="${this._handleCheckboxActionClick}">${inner}</div>`
}
```

Reply:
I agree this is a better approach - will update accordingly

Reply:
Updated and changed to two separate methods so that the consumer is in control of the layout only.